_posts/2020-05-18-mirror-kafka-eventhub.md

```diff
@@ -0,0 +1,219 @@
+
+![Azure Portal: Event Hub creation](/assets/images/posts/2020-05-18-azure-portal-eventhub.png)
+
+Having the Event Hub ready is just the first step.
+In order to configure Kafka Mirror Maker for connecting to the Event Hub, we need to get the corresponding connection string for being authenticated and authorized to write on the Event Hub itself.
+Just for simplicity, we can use the already provided `RootManageSharedAccessKey` policy that enable all the rights to manage, send and listen in the Event Hub namespace in which in the `testeh` lives.
+In a real scenario, we would like to create a policy for the mirroring part in order to enable the Kafka Mirror Maker producer just to write to the Event Hub while using a different policy for the application consuming the messages with just the listen right.
+
+The only needed information from the above policy is the connection string which has the following format:
+
+```
+Endpoint=sb://<eventhubs-namespace>.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=<access-key>
+```
+
+Where `<eventhubs-namespace>` has to be replaced with the actual Event Hub namespace you are going to use.
+
+![Azure Portal: Event Hub connection string](/assets/images/posts/2020-05-18-connectionstring-eventhub.png)
+
+### Configure Kafka Mirror Maker
+
+The first step is storing the Event Hub authentication info into a Kubernetes `Secret` in order to be referenced in the Strimzi `KafkaMirrorMaker` custom resource for enabling the producer part to connect to the Event Hub itself.
+The following snippet shows such a `Secret` where the `$ConnectionString` hasn't to be changed while the endpoint has to be customized with the actual Event Hub connection string; save this `Secret` in a file named `eventhubs-secret.yaml`
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: eventhubssecret
+type: Opaque
+stringData:
+  eventhubsuser: $ConnectionString
+  eventhubspassword: Endpoint=sb://<eventhubs-namespace>.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=<access-key>
+```
+
+Create the `Secret` on the Kubernetes cluster.
+
+```shell
+kubectl apply -f eventhubs-secret.yaml -n kafka
+```
+
+The Kafka Mirror Maker instance is deployed via the Strimzi cluster operator through a corresponding `KafkaMirrorMaker` resource as the following.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaMirrorMaker
+metadata:
+  name: my-mirror-maker
+spec:
+  version: 2.4.0
+  replicas: 1
+  consumer:
+    bootstrapServers: my-cluster-kafka-bootstrap:9092
+    groupId: my-source-group-id
+  producer:
+    bootstrapServers: <eventhubs-namespace>.servicebus.windows.net:9093
+    authentication:
+      type: plain
+      username: $ConnectionString
+      passwordSecret:
+        secretName: eventhubssecret
+        password: eventhubspassword
+    tls:
+      trustedCertificates: []
```

Comment:
I think this deserves some little explanation in the text - why is it used and how it works.